### ISSUE TITLE
Convert existing LA topics into study plans

### ISSUE THREAD

**Issue by nikitavoloboev** [learn-anything/learn-anything#1285]
Each topic currently on Learn Anything has resources and 'further to study' topics. Which is a valid study plan.

So we are already have ~2200 study plans with topics to use in the new version.

I am thinking of writing a script that would extract the links from existing LA and create markdown study plans from them. Perhaps some post script editing will be needed but I think it's a valid idea.

What do you think?

**Comment by nglgzz** [comment]
I don't think it's a good idea for the following reasons:

**Many of the resources are outdated.**
Most of the resources are about CS related topics (eg. machine learning, Javascript), those topics change very quickly and we haven't updated resources in a very long while.

**There's not enough structure.** 
Many resources cover the same content, just in different words. Study plans would lose their purpose if for each resource the user has to decide if it's worth to check out or not. I'm not referring to the quality nor user preference here; I mean that the user shouldn't ask themselves if the resource they are opening covers the same content from a previous one in the same study plan.

**It would set a bad example for contributors**
Study plans should have great quality and they should set the bar that other contributors would compare their study plans with. Based on the two points above I don't think these generated study plans would achieve this goal.

I do realize that creating study plans from scratch would take a very long time though. And our platform is not very useful without content. 

A better solution for this problem would be to get existing study plans that are actively maintained (there are lots of them on Reddit/Github, but they can be found anywhere else as well) and port them to LA.

**Comment by nikitavoloboev** [comment]
I am going to convert existing LA topics into guided plans and put them under my name then. 🙂



> there are lots of them on Reddit/Github

I agree. We should do that too.

**Comment by nikitavoloboev** [comment]
> Most of the resources are about CS related topics

That's not true. There's probably ~ 1300 technically related topics. The rest is light skills and everything else.

**Comment by nglgzz** [comment]
It's still most of them.

Also one point I didn't mention, a lot of topics contain just a couple of links.

**Comment by nglgzz** [comment]
You shouldn't be attached to something just because you put a lot of effort into it. I also put a lot of effort into old version of LA but in that state it wasn't very useful and changing it would take more effort than rewriting it, so I trashed the old code.

Many of those resources are not relevant anymore, and the effort you would spend finding and adapting the few decent study plans created by this script, would give greater results if you spent it searching for reddit wikis and converted those to study plans. 

They would have better quality as well. Those wikis are constantly maintained by a lot of people, while the resources on LA you added them at some point last year and never changed them after that. 

Also, nothing goes to waste. The same way we learned something from the old code, we learned something from those maps. And the connections between topics is still very useful.

**Comment by nikitavoloboev** [comment]
Basically need to write a script to parse

https://github.com/learn-anything/learn-anything/blob/2017/maps/maps-bak.json 

And then the existing mind maps too as we settile on the structure that our study guides will be using.

In the end we get ~ 2000 topics for initial release. As well as ~ 2000 study guides attached to each of the topics (with some cleanup). Perhaps those study guides can be released officialy by Learn Anything.
